processing/src/main/java/org/apache/druid/segment/filter/SpatialFilter.java

```diff
@@ -174,8 +173,18 @@ public DruidObjectPredicate<String> makeStringPredicate()
         if (input == null) {
           return DruidPredicateMatch.UNKNOWN;
         }
-        final float[] coordinate = SpatialDimensionRowTransformer.decode(input);
-        return DruidPredicateMatch.of(bound.contains(coordinate));
+        return DruidPredicateMatch.of(bound.containsObj(input));
+      };
+    }
+
+    @Override
+    public DruidObjectPredicate<Object> makeObjectPredicate()
+    {
+      return input -> {
+        if (input == null) {
+          return DruidPredicateMatch.UNKNOWN;
+        }
+        return DruidPredicateMatch.of(bound.containsObj(input));
```

Comment:
I wonder if `input` could be a `float[]` or not at this point - could it be?

Reply:
Spatial column is stored as String and input can be only String in this case.

Reply:
containsObj method is mainly added for extensibility perspective, it makes life easier to create object matchers and we can utilize the existing spatial filters.